### PR TITLE
Allow defining additional data passed to codestarts for project creation

### LINKED ISSUE #28372: Add codestart data option in CLI and Maven plugin
### Description

It would be nice extension codestart to have user customizable data. This is already possible in codestarts but not yet connected in the tooling.

It would behave like this:

```shell
quarkus create app -x=my-ext -d=my-ext-codestart.foo.bar=hello,my-ext-codestart.foo.hello=andy
```

The `my-ext-codestart` prefix is required by the codestart system to know the data is targeting this particular codestart data (and avoid conflicts)

The default value is in the codestart.yml:
```yaml
name: my-ext-codestart
language:
  base:
    data:
      foo:
        bar: baz
        hello: world
```

FYI: I created this issue because @gsmet has a need for this.

### PR DESCRIPTION
**Pull request by gsmet**
Fixes https://github.com/quarkusio/quarkus/issues/28372

Also fixes a few minor oversights in my previous patch in follow-up commits.

@ia3andy I tested things manually but I'm unsure how practical it would be to test the new `data` feature as we would need a specific codestart to test it actually works. I think it goes a bit beyond what I'm ready to invest in this feature but maybe you have a better idea?

### CODE REVIEW

**Comment by gsmet** [comment]
BTW, ideally, I would like to merge this before Tuesday evening so that it can get into 2.15.0.Final.

**Review by ia3andy — APPROVED** [review]
Cool!

Sent a test on Zulip

File: devtools/cli/src/main/java/io/quarkus/cli/common/DataOptions.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+import picocli.CommandLine;
+
+public class DataOptions {
+
+    public Map<String, String> data = new HashMap<>();
+
+    @CommandLine.Option(names = "--data", mapFallbackValue = "", description = "Additional data")
```
Comment (maxandersen) [inline devtools/cli/src/main/java/io/quarkus/cli/common/DataOptions.java:12]: What is the difference between --data and -D used in other commands to pass key/values in?

**Review by maxandersen — COMMENTED** [review]
I know it's closed but didn't see this pr before now so just want to grok why use --data when we already have -D used for other commands ?

File: devtools/cli/src/main/java/io/quarkus/cli/common/DataOptions.java (hunk 1 above)
Comment (maxandersen) [reply devtools/cli/src/main/java/io/quarkus/cli/common/DataOptions.java:12]: never mind - just realizing this happens at create time where -D already are present.

**Pull request merged**